### PR TITLE
Declarative action list sections

### PR DESCRIPTION
**Pull request by jasonrhodes**
Oftentimes you want to declaratively specify sections rather than pushing the items into separate arrays. This change lets you do
```js
actions={[
  {
    content: 'whatever',
    section: 1
  },
  {
    content: 'whatever 2',
    section: 2
  }
]}
```
and have it work exactly the same as
```js
sections={[
  [
    {
      content: 'whatever'
    }
  ],
  [
    {
      content: 'whatever 2'
    }
  ]
]}
```

### CODE REVIEW

File: packages/matchbox/src/components/ActionList/ActionList.js [hunk 1]
```diff
@@ -47,22 +49,32 @@ class ActionList extends Component {
     /**
       * Max height of list
       */
-    maxHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+    maxHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+    /**
+     * Group by key used to auto group actions into sections, defaults to "section"
+     */
+    groupByKey: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.symbol])
```
Comment (jasonrhodes) [inline packages/matchbox/src/components/ActionList/ActionList.js]: should only be string

File: packages/matchbox/src/components/ActionList/ActionList.js [hunk 2]
```diff
@@ -47,22 +49,32 @@ class ActionList extends Component {
...
 
-    let list = actions.length ? [actions] : [];
+    console.log('found actions', actions);
+
+    let list = actions.length ? groupByValues(actions, groupByKey) : [];
     if (sections.length) {
       list = list.concat(sections);
     }
 
+    console.log('grouped', list);
```
Comment (jasonrhodes) [inline packages/matchbox/src/components/ActionList/ActionList.js]: remove

File: packages/matchbox/src/components/ActionList/ActionList.js [hunk 3]
```diff
@@ -47,18 +49,29 @@ class ActionList extends Component {
...
+    maxHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+    /**
+     * Group by key used to auto group actions into sections, defaults to "section"
+     */
+    groupByKey: PropTypes.string
+  };
+
+  static defaultProps = {
+    maxHeight: 'none',
```
Comment (jonambas) [inline packages/matchbox/src/components/ActionList/ActionList.js]: i'm okay with removing this default prop, and defaulting in render instead

**Comment by jonambas** [comment]
Merged into #128

**Pull request closed without merging**